### ISSUE TITLE
False negatives superfluous-parens with more than one superfluous parenthesis

### ISSUE THREAD

**Issue by KotlinIsland** [PyCQA/pylint#5361]
### Current problem

```py
for (i) in [1]:  # superfluous-parens
    print(i)

for ((i)) in [1]:  # no warning
    print(i)
```

### Desired solution

warning on both

**Comment by jacobtylerwalls** [comment]
See [discussion](https://github.com/PyCQA/pylint/pull/5964#discussion_r835280442) -- we might be able to revert most of the changes from #4948 to solve this. Just check #4907 for any legitimate cases.

**Comment by clavedeluna** [comment]
Investigating this and trying to translate the context into an actionable. Is the new consensus that any test case **except  the one with startswith((** in `tests/functional/s/superfluous_parens.py` which contains `((` and does not report `superfluous-parens` should now do so?
